### PR TITLE
Latest oauth-proxy with bcrypt

### PR DESCRIPTION
**Pull request by csibbitt**
- Moving to latest oauth-proxy so we can stop using SHA-1 passwords
- Added required deps for bcrypt
- Set up RELATED_IMAGE handling

### CODE REVIEW

**Comment by csibbitt** [comment]
BTW, the previous commit passed our CI smoke tests: 
image

This one just passed linting, so just that build deps problem to deal with.

File: build/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,11 @@
...
...JhI7Cq0\nWm8DgLUpXgNlkmquvE7P2W5EAr2E5AqKQoDbfw/GiWdRvHWKeNGMRLnGI3QuoX3U\npAlXD7v13VdZxNydvpeypbf/AfRyrHRKhkUj3cU1pYkM3DNZE77C5JUe6/0nxbt4\nETUZBTgLgYJGP8c7PbkVnO6I/KgL1jw+7MW6Az8Ox+RXZLyGMVmbW/TMc8haJfKL\nMoUo3TVk8nPiUhoOC0/kI7j9ilFrBxBU5dUtF4ITAWc8xnG6jJs/IsvRpQARAQAB\ntChGZWRvcmEgRVBFTCAoOCkgPGVwZWxAZmVkb3JhcHJvamVjdC5vcmc+iQI4BBMB\nAgAiBQJc9877AhsPBgsJCAcDAgYVCAIJCgsEFgIDAQIeAQIXgAAKCRAh6kWrL4bW\noWagD/4xnLWws34GByVDQkjprk0fX7Iyhpm/U7BsIHKspHLL+Y46vAAGY/9vMvdE\n0fcr9Ek2Zp7zE1RWmSCzzzUgTG6BFoTG1H4Fho/7Z8BXK/jybowXSZfqXnTOfhSF\nalwDdwlSJvfYNV9MbyvbxN8qZRU1z7PEWZrIzFDDToFRk0R71zHpnPTNIJ5/YXTw\nNqU9OxII8hMQj4ufF11040AJQZ7br3rzerlyBOB+Jd1zSPVrAPpeMyJppWFHSDAI\nWK6x+am13VIInXtqB/Cz4GBHLFK5d2/IYspVw47Solj8jiFEtnAq6+1Aq5WH3iB4\nbE2e6z00DSF93frwOyWN7WmPIoc2QsNRJhgfJC+isGQAwwq8xAbHEBeuyMG8GZjz\nxohg0H4bOSEujVLTjH1xbAG4DnhWO/1VXLX+LXELycO8ZQTcjj/4AQKuo4wvMPrv\n9A169oETG+VwQlNd74VBPGCvhnzwGXNbTK/KH1+WRH0YSb+41flB3NKhMSU6dGI0\nSGtIxDSHhVVNmx2/6XiT9U/znrZsG5Kw8nIbbFz+9MGUUWgJMsd1Zl9R8gz7V9fp\nn7L7y5LhJ8HOCMsY/Z7/7HUs+t/A1MI4g7Q5g5UuSZdgi0zxukiWuCkLeAiAP4y7\nzKK4OjJ644NDcWCHa36znwVmkz3ixL8Q0auR15Oqq2BjR/fyog==\n=84m8\n-----END PGP PUBLIC KEY BLOCK-----" > /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8'
```
Comment (leifmadsen) [inline build/Dockerfile:4]: Only required for upstream I assume, correct?

**Review by leifmadsen — APPROVED** [review]
This all looks pretty good, and I saw a show (no tell) linked to me showing this working.

For downstream changes, I'm not entirely sure exactly how to deal with that. Either we can document what changes are required and make sure we perform those changes prior to the next release, or stage the changes downstream just prior to merging this for the upstream project.

Mostly I don't want to get into a situation where we forget to adjust the Dockerfile and metadata paths in the product pipeline.

File: build/Dockerfile (hunk 1 above)
Comment (leifmadsen) [reply build/Dockerfile:4]: Kind of a silly question, but I'm pretty sure the packages would be available from product pipeline repositories available to the build system.

Comment (csibbitt) [reply build/Dockerfile:4]: Yes, only upstream. I have another revision coming that checks for (I hope) upstream vs. downstream builds so we avoid polluting the downstream ones or requiring further Dockerfile templating or whatever.

Comment (leifmadsen) [reply build/Dockerfile:4]: OK thanks for confirming. Just to be clear, there are separate Dockerfiles here and in the downstream repository, and changes need to be made separately. The Dockerfile here is not imported and used in the midstream repository at all.

Path to the Dockerfile.in is at internal GL >> rhos-stf/service-telemetry-operator/-/blob/rhos-stf-1.5-rhel-8/distgit/containers/service-telemetry-operator/Dockerfile.in

Changes also required at rhos-stf/service-telemetry-operator/-/blob/rhos-stf-1.5-rhel-8/distgit/containers/service-telemetry-operator-bundle/render_vars.in to add the OAUTH_PROXY image path for the RELATED_IMAGES bits.

This is one of the more unfortunate bits of our import procedures :(

Comment (csibbitt) [reply build/Dockerfile:4]: Ah, thanks. I actually have an MR in place for the render_vars, but missed the Dockerfile. I'll add it; thanks.

**Comment by csibbitt** [comment]
I've solved the build deps problem.

On my openshift build environment inside the firewall, I see it using localdev-openstack-16-for-rhel-8-rpms as the repo for these packages:

```
python3-bcrypt    x86_64  3.1.6-2.el8ost localdev-openstack-16-for-rhel-8-rpms   45 k
```

Whereas upstream in the CI tests I see it using the injected EPEL repo:

```
python3-bcrypt         x86_64        3.1.6-2.el8.1           epel         44 k
```
https://github.com/infrawatch/service-telemetry-operator/actions/runs/4725219162/jobs/8383352481?pr=420#step:3:127

I'll update against master and then trigger the CIs test manually (comment triggers appear broken) to get full :heavy_check_mark:

**Pull request merged**